src/mscorlib/src/System/Globalization/CompareInfo.cs

```diff
@@ -170,12 +176,20 @@ public static CompareInfo GetCompareInfo(String name)
 
         public static unsafe bool IsSortable(char ch)
         {
+            if (CultureData.InvariantMode)
+            {
+                return true;
+            }
```

Comment:
I have spent some time before thinking about how we can implement the whole thing here. I have looked at the alternatives. for CultureInfo we already doing almost your idea mentioned here. but for other classes like CompareInfo/TextInfo I preferred the current implementation. to make it like what you suggested, I'll have to subclass CompareInfo/TextInfo and override all methods and then return such objects to the callers of such properties (like CultureInfo.CompareInfo, CultureInfo.TextInfo) . I dismissed this idea because a) we are not returning CompareInfo as we used to but we are returning a subclass of CompareInfo b) I was trying to avoid the unknown perf cost c) this will restrict us if we decided to expose a constructors for such classes in the future d) we'll have to use most of the existing code in CompareInfo anyway. we can refactor the code for this one though.